packages/shared/components/Select/Select.tsx

```diff
@@ -59,7 +63,13 @@ export function SelectAsync(props: AsyncProps) {
   );
 }
 
-export const StyledSelect = styled.div`
+interface StyledSelectBaseProps {
+  hasError?: boolean;
+}
+
+export type StyledSelectProps = StyledSelectBaseProps & SpaceProps & WidthProps;
+
+export const StyledSelect = styled.div<PropsWithTheme<StyledSelectProps>>`
```

Comment:
```suggestion
export const StyledSelect = styled.div<StyledSelectProps>`
```
Theme is injected by `styled.div`.